src/app/workflow/view/view.component.ts

```diff
@@ -89,6 +93,25 @@ export class ViewWorkflowComponent extends View implements OnInit {
     });
   }
 
+  snapshotVersion(): void {
+    // Create a new version temporarily with the frozen bit set. We assume the version is
+    // not already a snapshot since the UI controls shouldn't be available for
+    // snapshotted versions.
+    var snapshot = { ...this.version };
+    snapshot['frozen'] = true;
+    this.workflowsService.updateWorkflowVersion(this.workflow.id, [snapshot]).subscribe(
+      result => {
+        this.alertService.detailedSuccess('Snapshot successfully created!');
+        this.refreshService.refreshWorkflow();
+      },
+      error => {
+        if (error) {
```

Comment:
You should display an error even if `error` is not set -- call `simpleError` if not set.